src/polycubed/src/port.cpp

```diff
@@ -85,11 +87,9 @@ void Port::set_peer(const std::string &peer) {
     if (peer_ == peer) {
       return;
     }
+    peer_ = peer;
```

Comment:
`peer_` is updated after `ServiceController::set_port_peer(*this, peer);` because it is only after that function finishes that the peer has actually been updated.

There are some problems of moving it there.
1. The logic to remove a previous connection is broken, it'll try to remove a new non existing connection while the old one will remain there. https://github.com/polycube-network/polycube/blob/master/src/polycubed/src/service_controller.cpp#L231
2. If `ServiceController::set_port_peer(*this, peer);` raises an exception the peer would be configured to a wrong value.